catroid/src/androidTest/java/org/catrobat/catroid/test/embroidery/EmbroideryFileExporterTest.java

```diff
@@ -0,0 +1,100 @@
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.embroidery;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.v4.content.FileProvider;
+
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.embroidery.EmbroideryFileExporter;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class EmbroideryFileExporterTest {
+
+	private Context context;
+	private EmbroideryFileExporter embroideryFileExporter;
+	private String filename = EmbroideryFileExporter.class.getName() + ".dst";
+
+	@Before
+	public void setUp() {
+		context = Mockito.mock(Context.class);
+		Mockito.when(context.getPackageManager()).thenReturn(InstrumentationRegistry.getTargetContext().getPackageManager());
+		Mockito.when(context.getPackageName()).thenReturn(InstrumentationRegistry.getTargetContext().getPackageName());
+		embroideryFileExporter = new EmbroideryFileExporter();
+	}
+
+	@Test
+	@Category({Level.Smoke.class})
+	public void testShareNonExistentFile() {
+		File file = new File(Constants.CACHE_DIR, filename);
+		file.delete();
+		embroideryFileExporter.shareEmbroideryFile(context, file);
+		Mockito.verify(context, Mockito.never()).startActivity(Mockito.any(Intent.class));
+	}
+
+	@Test
+	@Category({Level.Smoke.class})
+	public void testShareSimpleFile() throws IOException {
+		File file = new File(Constants.CACHE_DIR, filename);
+		file.createNewFile();
+		embroideryFileExporter.shareEmbroideryFile(context, file);
+
+		ArgumentCaptor<Intent> captor = ArgumentCaptor.forClass(Intent.class);
+		Mockito.verify(context, Mockito.times(1)).startActivity(captor.capture());
+
+		Intent chooserIntent = captor.getValue();
+		Intent shareIntent = chooserIntent.getParcelableExtra(Intent.EXTRA_INTENT);
+
+		File dstFile = new File(Constants.CACHE_DIR, filename);
+		Uri uriForFile = FileProvider.getUriForFile(context, context.getPackageName() + ".fileProvider", dstFile);
+
+		Intent compareShareIntent = new Intent(Intent.ACTION_SEND, uriForFile);
```

Comment:
can you please rename those intents to expectedWhateverIntent and actualWhateverIntent.
its really hard to read the way it is now, and really hard to comprehend which part is test setup, which is produced by production code.

also please move the assertion to the respective blocks where you create your expectedIntents.